ibis/backends/datafusion/compiler/values.py

```diff
@@ -710,3 +713,26 @@ def _if_else(op, *, bool_expr, true_expr, false_null_expr, **_):
 @translate_val.register(ops.NotNull)
 def _not_null(op, *, arg, **_):
     return sg.not_(arg.is_(NULL))
+
+
+@translate_val.register(ops.ArrayColumn)
+def array_column(op, *, cols, **_):
+    return F.make_array(*cols)
+
+
+@translate_val.register(ops.ArrayRepeat)
+def array_repeat(op, *, arg, times, **_):
+    return F.flatten(F.array_repeat(arg, times))
+
+
+@translate_val.register(ops.ArrayConcat)
+def array_concat(op, *, arg, **_):
+    return F.array_concat(*arg)
+
+
+@translate_val.register(ops.ArrayPosition)
+def array_position(op, *, arg, other, **_):
+    return sg.exp.Coalesce(
+        this=F.array_position(arg, other),
+        expressions=sg.exp.Literal(this="0", is_string=False),
```

Comment:
You should be able to use `sg.exp.convert(0)` here.

Reply:
You might also be able to avoid the `Coalesce` construction and use `F.coalesce` instead, then you wouldn't need the convert call.

Reply:
Does `expressions` need to be a `list`?

Reply:
Thanks! I went with `F.coalesce(F.array_position(arg, other), 0)`